python/taichi/lang/misc.py

```diff
@@ -346,6 +346,9 @@ def init(arch=None,
     # Check version for users every 7 days if not disabled by users.
     _version_check.start_version_check_thread()
 
+    # save current working directory since it may be changed by the vulkan backend
+    current_dir = os.getcwd()
```

Comment:
nit: Do we know which exactly step changed WD?

Reply:
Also, I think this is a workaround, not a fix. Consider writing the comment as `FIXME(issue-id): ...`